[Tech Debt]: Serialize `aws_finspace_kx_*` acceptance tests

### Description

The FinSpace managed kdb resource acceptance tests all require a kdb environment to execute. The default quota for `Total kdb environments` is 1, so to avoid `LimitExceeded` exceptions these tests should be serialized.

Example parallel test failure:
```
=== RUN   TestAccFinSpaceKxEnvironment_description
=== PAUSE TestAccFinSpaceKxEnvironment_description
=== CONT  TestAccFinSpaceKxEnvironment_description
    kx_environment_test.go:141: Step 1/2 error: Error running apply: exit status 1
        Error: creating Amazon FinSpace Kx Environment (tf-acc-test-736091273136364188): operation error finspace: CreateKxEnvironment, exceeded maximum number of attempts, 25, https response error StatusCode: 400, RequestID: 0a261b33-8fef-4655-aa52-85254aee3aef, LimitExceededException: Create environment failed due to capacity limit reached, environment name: tf-acc-test-736091273136364188
          with aws_finspace_kx_environment.test,
          on terraform_plugin_test.tf line 6, in resource "aws_finspace_kx_environment" "test":
           6: resource "aws_finspace_kx_environment" "test" {
--- FAIL: TestAccFinSpaceKxEnvironment_description (614.08s)
FAIL
```

### References

- https://docs.aws.amazon.com/finspace/latest/userguide/finspace-quotas.html


### Would you like to implement a fix?

None

--- a/docs/acc-test-environment-variables.md
+++ b/docs/acc-test-environment-variables.md
@@ -67,6 +67,7 @@ Environment variables (beyond standard AWS Go SDK ones) used by acceptance testi
 | `EC2_SECURITY_GROUP_RULES_PER_GROUP_LIMIT` | EC2 Quota for Rules per Security Group. Defaults to 50. **DEPRECATED:** Can be augmented or replaced with Service Quotas lookup. |
 | `EVENT_BRIDGE_PARTNER_EVENT_BUS_NAME` | Amazon EventBridge partner event bus name. |
 | `EVENT_BRIDGE_PARTNER_EVENT_SOURCE_NAME` | Amazon EventBridge partner event source name. |
+| `FINSPACE_MANAGED_KX_LICENSE_ENABLED` | Enables tests requiring a license to provision managed KX resources. |
 | `GCM_API_KEY` | API Key for Google Cloud Messaging in Pinpoint and SNS Platform Application testing. |
 | `GITHUB_TOKEN` | GitHub token for CodePipeline testing. |
 | `GLOBALACCERATOR_BYOIP_IPV4_ADDRESS` | IPv4 address from a BYOIP CIDR of AWS Account used for testing Global Accelerator's BYOIP accelerator. |

--- a/internal/service/finspace/finspace_test.go
+++ b/internal/service/finspace/finspace_test.go
@@ -1,0 +1,55 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package finspace_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccFinSpace_serial(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]map[string]func(t *testing.T){
+		"KxCluster": {
+			"autoScaling":          testAccKxCluster_autoScaling,
+			"basic":                testAccKxCluster_basic,
+			"cacheConfigurations":  testAccKxCluster_cacheConfigurations,
+			"code":                 testAccKxCluster_code,
+			"commandLineArgs":      testAccKxCluster_commandLineArgs,
+			"database":             testAccKxCluster_database,
+			"description":          testAccKxCluster_description,
+			"disappears":           testAccKxCluster_disappears,
+			"executionRole":        testAccKxCluster_executionRole,
+			"initializationScript": testAccKxCluster_initializationScript,
+			"multiAZ":              testAccKxCluster_multiAZ,
+			"rdb":                  testAccKxCluster_rdb,
+			"tags":                 testAccKxCluster_tags,
+		},
+		"KxDatabase": {
+			"basic":       testAccKxDatabase_basic,
+			"description": testAccKxDatabase_description,
+			"disappears":  testAccKxDatabase_disappears,
+			"tags":        testAccKxDatabase_tags,
+		},
+		"KxEnvironment": {
+			"basic":          testAccKxEnvironment_basic,
+			"customDNS":      testAccKxEnvironment_customDNS,
+			"description":    testAccKxEnvironment_description,
+			"disappears":     testAccKxEnvironment_disappears,
+			"tags":           testAccKxEnvironment_tags,
+			"transitGateway": testAccKxEnvironment_transitGateway,
+			"updateName":     testAccKxEnvironment_updateName,
+		},
+		"KxUser": {
+			"basic":      testAccKxUser_basic,
+			"disappears": testAccKxUser_disappears,
+			"tags":       testAccKxUser_tags,
+			"updateRole": testAccKxUser_updateRole,
+		},
+	}
+
+	acctest.RunSerialTests2Levels(t, testCases, 0)
+}

--- a/internal/service/finspace/kx_cluster_test.go
+++ b/internal/service/finspace/kx_cluster_test.go
@@ -22,7 +22,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-func TestAccFinSpaceKxCluster_basic(t *testing.T) {
+func testAccKxCluster_basic(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -32,7 +32,7 @@ func TestAccFinSpaceKxCluster_basic(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -58,7 +58,7 @@ func TestAccFinSpaceKxCluster_basic(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_disappears(t *testing.T) {
+func testAccKxCluster_disappears(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -68,7 +68,7 @@ func TestAccFinSpaceKxCluster_disappears(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -89,7 +89,7 @@ func TestAccFinSpaceKxCluster_disappears(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_description(t *testing.T) {
+func testAccKxCluster_description(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -99,7 +99,7 @@ func TestAccFinSpaceKxCluster_description(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -119,7 +119,7 @@ func TestAccFinSpaceKxCluster_description(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_database(t *testing.T) {
+func testAccKxCluster_database(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -129,7 +129,7 @@ func TestAccFinSpaceKxCluster_database(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -149,7 +149,7 @@ func TestAccFinSpaceKxCluster_database(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_cacheConfigurations(t *testing.T) {
+func testAccKxCluster_cacheConfigurations(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -159,7 +159,7 @@ func TestAccFinSpaceKxCluster_cacheConfigurations(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -179,7 +179,7 @@ func TestAccFinSpaceKxCluster_cacheConfigurations(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_code(t *testing.T) {
+func testAccKxCluster_code(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -190,7 +190,7 @@ func TestAccFinSpaceKxCluster_code(t *testing.T) {
 	resourceName := "aws_finspace_kx_cluster.test"
 	codePath := "test-fixtures/code.zip"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -209,7 +209,7 @@ func TestAccFinSpaceKxCluster_code(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_multiAZ(t *testing.T) {
+func testAccKxCluster_multiAZ(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -219,7 +219,7 @@ func TestAccFinSpaceKxCluster_multiAZ(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -239,7 +239,7 @@ func TestAccFinSpaceKxCluster_multiAZ(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_rdb(t *testing.T) {
+func testAccKxCluster_rdb(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -249,7 +249,7 @@ func TestAccFinSpaceKxCluster_rdb(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -269,7 +269,7 @@ func TestAccFinSpaceKxCluster_rdb(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_executionRole(t *testing.T) {
+func testAccKxCluster_executionRole(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -279,7 +279,7 @@ func TestAccFinSpaceKxCluster_executionRole(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -299,7 +299,7 @@ func TestAccFinSpaceKxCluster_executionRole(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_autoScaling(t *testing.T) {
+func testAccKxCluster_autoScaling(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -309,7 +309,7 @@ func TestAccFinSpaceKxCluster_autoScaling(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -329,7 +329,7 @@ func TestAccFinSpaceKxCluster_autoScaling(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_initializationScript(t *testing.T) {
+func testAccKxCluster_initializationScript(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -342,7 +342,7 @@ func TestAccFinSpaceKxCluster_initializationScript(t *testing.T) {
 	codePath := "test-fixtures/code.zip"
 	initScriptPath := "code/helloworld.q"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -361,7 +361,7 @@ func TestAccFinSpaceKxCluster_initializationScript(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_commandLineArgs(t *testing.T) {
+func testAccKxCluster_commandLineArgs(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -371,7 +371,7 @@ func TestAccFinSpaceKxCluster_commandLineArgs(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -392,7 +392,7 @@ func TestAccFinSpaceKxCluster_commandLineArgs(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxCluster_tags(t *testing.T) {
+func testAccKxCluster_tags(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -402,7 +402,7 @@ func TestAccFinSpaceKxCluster_tags(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_cluster.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)

--- a/internal/service/finspace/kx_cluster_test.go
+++ b/internal/service/finspace/kx_cluster_test.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
 	"testing"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -22,6 +23,15 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
+func testAccPreCheckManagedKxLicenseEnabled(t *testing.T) {
+	if os.Getenv("FINSPACE_MANAGED_KX_LICENSE_ENABLED") == "" {
+		t.Skip(
+			"Environment variable FINSPACE_MANAGED_KX_LICENSE_ENABLED is not set. " +
+				"Certain managed KX resources require the target account to have an active " +
+				"license. Set the environment variable to any value to enable these tests.")
+	}
+}
+
 func testAccKxCluster_basic(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
@@ -36,6 +46,7 @@ func testAccKxCluster_basic(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -72,6 +83,7 @@ func testAccKxCluster_disappears(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -103,6 +115,7 @@ func testAccKxCluster_description(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -133,6 +146,7 @@ func testAccKxCluster_database(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -163,6 +177,7 @@ func testAccKxCluster_cacheConfigurations(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -194,6 +209,7 @@ func testAccKxCluster_code(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -223,6 +239,7 @@ func testAccKxCluster_multiAZ(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -253,6 +270,7 @@ func testAccKxCluster_rdb(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -283,6 +301,7 @@ func testAccKxCluster_executionRole(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -313,6 +332,7 @@ func testAccKxCluster_autoScaling(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -346,6 +366,7 @@ func testAccKxCluster_initializationScript(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -375,6 +396,7 @@ func testAccKxCluster_commandLineArgs(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
@@ -406,6 +428,7 @@ func testAccKxCluster_tags(t *testing.T) {
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+			testAccPreCheckManagedKxLicenseEnabled(t)
 		},
 		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,

--- a/internal/service/finspace/kx_database_test.go
+++ b/internal/service/finspace/kx_database_test.go
@@ -22,7 +22,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-func TestAccFinSpaceKxDatabase_basic(t *testing.T) {
+func testAccKxDatabase_basic(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -32,7 +32,7 @@ func TestAccFinSpaceKxDatabase_basic(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_database.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -57,7 +57,7 @@ func TestAccFinSpaceKxDatabase_basic(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxDatabase_disappears(t *testing.T) {
+func testAccKxDatabase_disappears(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -67,7 +67,7 @@ func TestAccFinSpaceKxDatabase_disappears(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_database.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -88,7 +88,7 @@ func TestAccFinSpaceKxDatabase_disappears(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxDatabase_description(t *testing.T) {
+func testAccKxDatabase_description(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -98,7 +98,7 @@ func TestAccFinSpaceKxDatabase_description(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_database.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -125,7 +125,7 @@ func TestAccFinSpaceKxDatabase_description(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxDatabase_tags(t *testing.T) {
+func testAccKxDatabase_tags(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -135,7 +135,7 @@ func TestAccFinSpaceKxDatabase_tags(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_database.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)

--- a/internal/service/finspace/kx_environment_test.go
+++ b/internal/service/finspace/kx_environment_test.go
@@ -22,7 +22,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-func TestAccFinSpaceKxEnvironment_basic(t *testing.T) {
+func testAccKxEnvironment_basic(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -33,7 +33,7 @@ func TestAccFinSpaceKxEnvironment_basic(t *testing.T) {
 	resourceName := "aws_finspace_kx_environment.test"
 	kmsKeyResourceName := "aws_kms_key.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -59,7 +59,7 @@ func TestAccFinSpaceKxEnvironment_basic(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_disappears(t *testing.T) {
+func testAccKxEnvironment_disappears(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -69,7 +69,7 @@ func TestAccFinSpaceKxEnvironment_disappears(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -90,7 +90,7 @@ func TestAccFinSpaceKxEnvironment_disappears(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_updateName(t *testing.T) {
+func testAccKxEnvironment_updateName(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -101,7 +101,7 @@ func TestAccFinSpaceKxEnvironment_updateName(t *testing.T) {
 	rName2 := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -128,7 +128,7 @@ func TestAccFinSpaceKxEnvironment_updateName(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_description(t *testing.T) {
+func testAccKxEnvironment_description(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -138,7 +138,7 @@ func TestAccFinSpaceKxEnvironment_description(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -165,7 +165,7 @@ func TestAccFinSpaceKxEnvironment_description(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_customDNS(t *testing.T) {
+func testAccKxEnvironment_customDNS(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -175,7 +175,7 @@ func TestAccFinSpaceKxEnvironment_customDNS(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -208,7 +208,7 @@ func TestAccFinSpaceKxEnvironment_customDNS(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_transitGateway(t *testing.T) {
+func testAccKxEnvironment_transitGateway(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -218,7 +218,7 @@ func TestAccFinSpaceKxEnvironment_transitGateway(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -240,7 +240,7 @@ func TestAccFinSpaceKxEnvironment_transitGateway(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxEnvironment_tags(t *testing.T) {
+func testAccKxEnvironment_tags(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -250,7 +250,7 @@ func TestAccFinSpaceKxEnvironment_tags(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	resourceName := "aws_finspace_kx_environment.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)

--- a/internal/service/finspace/kx_user_test.go
+++ b/internal/service/finspace/kx_user_test.go
@@ -22,7 +22,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-func TestAccFinSpaceKxUser_basic(t *testing.T) {
+func testAccKxUser_basic(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -33,7 +33,7 @@ func TestAccFinSpaceKxUser_basic(t *testing.T) {
 	userName := sdkacctest.RandString(sdkacctest.RandIntRange(1, 50))
 	resourceName := "aws_finspace_kx_user.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -58,7 +58,7 @@ func TestAccFinSpaceKxUser_basic(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxUser_disappears(t *testing.T) {
+func testAccKxUser_disappears(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -69,7 +69,7 @@ func TestAccFinSpaceKxUser_disappears(t *testing.T) {
 	userName := sdkacctest.RandString(sdkacctest.RandIntRange(1, 50))
 	resourceName := "aws_finspace_kx_user.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -90,7 +90,7 @@ func TestAccFinSpaceKxUser_disappears(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxUser_updateRole(t *testing.T) {
+func testAccKxUser_updateRole(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
 	}
@@ -101,7 +101,7 @@ func TestAccFinSpaceKxUser_updateRole(t *testing.T) {
 	userName := sdkacctest.RandString(sdkacctest.RandIntRange(1, 50))
 	resourceName := "aws_finspace_kx_user.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
@@ -126,7 +126,7 @@ func TestAccFinSpaceKxUser_updateRole(t *testing.T) {
 	})
 }
 
-func TestAccFinSpaceKxUser_tags(t *testing.T) {
+func testAccKxUser_tags(t *testing.T) {
 	ctx := acctest.Context(t)
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
@@ -137,7 +137,7 @@ func TestAccFinSpaceKxUser_tags(t *testing.T) {
 	userName := sdkacctest.RandString(sdkacctest.RandIntRange(1, 50))
 	resourceName := "aws_finspace_kx_user.test"
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(ctx, t)
 			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)

